### ISSUE TITLE
Adding onChanged method in SearchDelegate class

### ISSUE THREAD

**Issue by swagftw** [flutter/flutter#57425]
## Use case
Adding an onChanged method for query in search delegate abstract class will be so much handy I guess. If I already have the list to be shown in the build results section then why should i use streams and stream builder as they are pretty heavy widgets to be used. Adding on query changed method can help us filter list and show in result as soon as query changes.

## Proposal
Is there any workaround for what I am trying to point out above? 
Or flutter should give access to the textfield just like textformfield so that onchanged method can be used to show results.

**Comment by darkfusion90** [comment]
Yes please. This should be implemented. A probable solution can be allowing the users to pass a `onChanged` callback and use that on query TextField's `onChanged` or a `controller` for the query TextField

**Comment by darkfusion90** [comment]
Yes please. This should be implemented. A probable solution can be allowing the users to pass a `onChanged` callback and use that on query TextField's `onChanged` or a `controller` for the query TextField

**Comment by zaidrix** [comment]
I have done it through a simple workaround

Simply after your database call ... just add this line

`query = query`
